### ISSUE TITLE
Strange behaviour with option flags

### ISSUE THREAD

**Issue by mikespokefire** [dry-rb/dry-cli#46]
Hi hanami-cli maintainers,

First of all, thank you for making a CLI framework which works well for what we have been doing! The command registry and commands as classes have been working really well for us.

We wanted to add aliases for some common options that are used throughout the CLI that we are building, they aren't documented as being available on options, but they seemed to work in a quick test when hacking around with the code. However we've seen some strange behaviour with options in general, it looks to be a problem with some pattern matching somewhere?

Given the following CLI structure:

```ruby
require 'hanami/cli'

module MyCommands
  extend Hanami::CLI::Registry

  class TheCommand < Hanami::CLI::Command
    option :aaa, aliases: ['a'], default: 'aaa-default', desc: 'aaa'
    option :abc, default: 'abc-default', desc: 'abc'

    def call(**opts)
      puts opts.inspect
    end
  end

  register 'the-command', TheCommand
end

cli = Hanami::CLI.new(MyCommands)
cli.call
```

When running the following: `bundle exec scratch.rb the-command -a hello` then we get the following error:

```
Error: "the-command" was called with arguments "-a hello"
```

However if we remove the first option

```ruby
require 'hanami/cli'

module MyCommands
  extend Hanami::CLI::Registry

  class TheCommand < Hanami::CLI::Command
    option :aaa, aliases: ['a'], default: 'aaa-default', desc: 'aaa'

    def call(**opts)
      puts opts.inspect
    end
  end

  register 'the-command', TheCommand
end

cli = Hanami::CLI.new(MyCommands)
cli.call
```

Then everything seems to work. Running `bundle exec scratch.rb the-command -a hello` we get the following (expected) output:

```
{:aaa=>"hello"}
```

I think this is by accident though, since if I remove the alias then the shorthand option...

```ruby
require 'hanami/cli'

module MyCommands
  extend Hanami::CLI::Registry

  class TheCommand < Hanami::CLI::Command
    option :aaa, default: 'aaa-default', desc: 'aaa'

    def call(**opts)
      puts opts.inspect
    end
  end

  register 'the-command', TheCommand
end

cli = Hanami::CLI.new(MyCommands)
cli.call
```

Then everything seems to work, even though this time I don't think it should. Running `bundle exec scratch.rb the-command -a hello` we get the following (unexpected) output:

```
{:aaa=>"hello"}
```

Then if I add back in the second option you get the following behaviour:

```ruby
require 'hanami/cli'

module MyCommands
  extend Hanami::CLI::Registry

  class TheCommand < Hanami::CLI::Command
    option :aaa, default: 'aaa-default', desc: 'aaa'
    option :abc, default: 'abc-default', desc: 'abc'

    def call(**opts)
      puts opts.inspect
    end
  end

  register 'the-command', TheCommand
end

cli = Hanami::CLI.new(MyCommands)
cli.call
```

Then if we run the same command again `bundle exec scratch.rb the-command -a hello` we get an error again:

```
Error: "the-command" was called with arguments "-a hello"
```

Many thanks,
Michael

**Comment by solnic** [comment]
@mikespokefire why did you close this? did it turn out to be an invalid issue?

**Comment by mikespokefire** [comment]
Not invalid, I just had no feedback for 7 months. As a result of this and
some other non-related issues we've moved our CLI framework over to
something else. I was merely cleaning up some issues I've had open for a
while 🙂.

Happy to reopen it if you want to track it as an issue?

**Comment by solnic** [comment]
@mikespokefire ok thanks, I re-opened

**Comment by IvanShamatov** [comment]
Thanks for your report, @mikespokefire. A year later I started working on it. Though, initial behaviour was because some unexpected behaviour of optparse lib, I found more things to fix. Fixes will be in 0.6 release

**Comment by IvanShamatov** [comment]
https://github.com/dry-rb/dry-cli/pull/68
